dask/blockwise.py

```diff
@@ -1412,6 +1412,60 @@ def _unique_dep(dep, ind):
     return dep + "_" + "_".join(str(i) for i in list(ind))
 
 
+def _can_fuse_annotations(a: dict | None, b: dict | None) -> bool:
+    """
+    Treat the special annotation keys, as fusable since we can apply simple
+    rules to capture their intent in a fused layer.
+    """
+    if a == b:
+        return True
+
+    if dask.config.get("optimization.fuse.blockwise-annotations") is False:
+        return False
```

Comment:
I'm wondering why this would need to be configurable. I'm a bit hesitant to add a new config option just for this

Reply:
A hypothetical example: if a user wanted to mark a particular operation as needing some resource like GPUs, this can result in a whole chain of blockwise operations before and after it needing that resource. In some case that might be what they want (task fusion is good!) but in other cases it might not be (I don't want to constrain all of these tasks to be on GPU workers!).

Reply:
Would one want to just disable blockwise optimizations altogether in that case?

Reply:
I suppose you could disable all graph optimizations in that case, though I'm unaware of an option for disabling *just* blockwise annotations.

Reply:
Okay, I see. This still seems complicated, which is unfortunate, but I see your motivation for adding it. I wouldn't mind seeing us revisiting this later if we refactor the optimization / annotation systems. 

Thoughts on using `optimization.annotations.fuse` for the config option? This name seems slightly more generic and the `optimization.fuse` section has to do with low-level task fusion separate from blockwise

Reply:
What about `optimization.annotations.fuse-blockwise`? I was trying to scope the name to just refer to blockwise, since this doesn't really affect annotations of any other layers.

Reply:
Do we anticipate possibly having the same need with other layers in the future? If so, I'd prefer `optimization.annotations.fuse` as a more generic setting we can reuse

Reply:
Sure, I have no real objections. The same rules described here could also be applied to other layers as well, even if we're not doing it now.

Reply:
Okay, done